litmus-portal/frontend/src/views/CreateWorkflow/ChooseWorkflow/index.tsx

```diff
@@ -183,6 +187,10 @@ const ChooseWorkflow: React.FC<ChooseWorkflowProps> = ({ isEditable }) => {
             <ButtonFilled
               data-cy="EditWorkflowButton"
               onClick={() => {
+                setPrevWorkflowData({
+                  workflowName: workflowDetails.workflowName,
```

Comment:
Why are we saving the name and desc changes onClose?

Reply:
We aren't saving them. The name and desc are being restored to what it was before the modal was opened. If the user types in a workflow name and then closes the modal, without clicking the save button, then that name won't persist.

Reply:
Can we instead just do an if check

if save not clicked set workflowName to 'Personal Workflow Name', that'd save adding a new state variable. Or you could use another approach like

let name = ''
handleChange -> name = ${name} (event.target.value) Thus the name would keep on appending letter by letter and once save is clicked if name !== '' then setName(name) else don't (the name remains the same)